File: Lab/app/src/main/java/id/ac/ui/cs/mobileprogramming/mutiarahmatun/helloworld/MainActivity.java
package id.ac.ui.cs.mobileprogramming.mutiarahmatun.helloworld;

import android.os.Bundle;
import android.widget.EditText;
import android.view.View;
import android.widget.RadioButton;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;


public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    private EditText mTemperatureInput;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTemperatureInput = (EditText) findViewById(R.id.main_temperature_input);

        //setting click listener to convert button
        findViewById(R.id.main_btn_convert).setOnClickListener(this);
    }
    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.main_btn_convert:
                RadioButton celsiusButton = (RadioButton) findViewById(R.id.main_radio_celsius);
                RadioButton fahrenheitButton = (RadioButton) findViewById(R.id.main_radio_fahr);
                if (mTemperatureInput.getText().length() == 0) {
                    Toast.makeText(MainActivity.this, getString(R.string.err_enter_valid_number),
                            Toast.LENGTH_LONG).show();
                    return;
                }

                float inputValue = Float.parseFloat(mTemperatureInput.getText().toString());
                if (celsiusButton.isChecked()) {
                    mTemperatureInput.setText(String
                            .valueOf(ConverterUtil.convertFahrenheitToCelcius(inputValue)));
                    celsiusButton.setChecked(false);
                    fahrenheitButton.setChecked(true);
                } else {
                    mTemperatureInput.setText(String
                            .valueOf(ConverterUtil.convertCelciusToFahrenheit(inputValue)));
                    fahrenheitButton.setChecked(false);
                    celsiusButton.setChecked(true);
                }

                break;
        }
    }
}

File: Lab-3/app/src/main/java/id/ac/ui/cs/mobileprogramming/mutiarahmatun/helloworld/MainActivity.kt
package id.ac.ui.cs.mobileprogramming.mutiarahmatun.helloworld

import android.os.Bundle
import android.os.Handler
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*


class MainActivity : AppCompatActivity() {
    private var seconds = 0
    private var running = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        runTimer()
        exitButton.setOnClickListener {
            super.onBackPressed()
        }
    }
    override fun onBackPressed() {
        Toast.makeText(
            this@MainActivity,
            "Click Exit to out from this app",
            Toast.LENGTH_LONG
        ).show()
    }
    private fun runTimer() {
        val timeView = findViewById(R.id.time_view) as TextView
        val handler = Handler()
        handler.post(object : Runnable {
            override fun run() {
                val hours = seconds / 3600
                val minutes = seconds % 3600 / 60
                val secs = seconds % 60
                val time: String = java.lang.String.format(
                    Locale.getDefault(),
                    "%d:%02d:%02d", hours, minutes, secs
                )
                timeView.text = time
                if (running) {
                    seconds++
                }
                handler.postDelayed(this, 1000)
            }
        })
    }

    //Start the stopwatch running when the Start button is clicked.
    fun onClickStart(view: View?) {
        running = true
    }

    //Stop the stopwatch running when the Stop button is clicked.
    fun onClickStop(view: View?) {
        running = false
    }

    //Reset the stopwatch when the Reset button is clicked.
    fun onClickReset(view: View?) {
        running = false
        seconds = 0
    }
}